merkledag/utils/utils.go

```diff
@@ -123,23 +153,29 @@ func rmLink(ctx context.Context, ds dag.DAGService, root *dag.Node, path []strin
 		return root, nil
 	}
 
-	nd, err := root.GetLinkedNode(ctx, ds, path[0])
+	nd, err := root.GetLinkedNode(ctx, e.tmp, path[0])
+	if err == dag.ErrNotFound {
+		nd, err = root.GetLinkedNode(ctx, e.src, path[0])
+	}
+
```

Comment:
i added the space purposely because i felt it made it more readable, do these things and then *in either case* check the error.